docs/5.x/catalog.md

```diff
@@ -0,0 +1,147 @@
+    needs to have `tiller` server (Helm's server component) running though.
+
+Both  `tele version` and `gravity version` commands report the embedded Helm
+version:
+
+```bsh
+$ tele version
+Edition:      open-source
+Version:      5.4.0
+Git Commit:   971d93b06da08c3b277fc033c1c4fdca55a0ec6b
+Helm Version: v2.12
+```
+
+### Build an Application Image
+
+For this example we will be using a [sample Helm chart](https://github.com/helm/helm/tree/master/docs/examples/alpine).
+This chart spins up a single pod of Alpine Linux:
+
+```bsh
+$ tree alpine
+alpine
+├── Chart.yaml
+├── README.md
+├── templates
+│   ├── alpine-pod.yaml
+│   └── _helpers.tpl
+└── values.yaml
+```
+
+Before building an application image, we need to make sure that the pod's
+image reference includes a registry template variable which can be set to
+an appropriate registry during installation:
+
+```bsh
+image: "{{ .Values.image.registry }}{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+```
+
+We can now use `tele` to build an application image from this chart:
+
+```bsh
+$ tele build alpine
+```
+
+The result is a tarball `alpine-0.1.0.tar` which includes a packaged Helm
+chart and the Alpine image layers.
+
+### Install a Release
+
+To deploy an application image, transfer it onto a cluster node and execute
+the install command:
+
+```bsh
+$ gravity app install alpine-0.1.0.tar
+```
+
+When executed inside a Gravity cluster, it will automatically push the
+application to the local cluster controller which will keep the images
+synchonized with the local Docker registries.
+
+When deploying into a generic Kubernetes cluster, the install command needs
+to know where to push the images and where the chart resources should pull
+them from:
+
+```bsh
+$ gravity app install alpine-0.1.0.tar \
+    --registry 10.103.27.132:5000 \
+    --set image.registry=10.103.27.132:5000/
+```
+
+Gravity manages application lifecycle using the embedded Helm which
+means that it creates a "release" for each deployed application image. The
+same application image can be installed into a cluster multiple times,
+and a new release will be created to track each installation.
+
+To view all currently deployed releases, run:
+
+```bsh
+$ gravity app ls
+Release         Status      Chart          Revision  Namespace  Updated
+falling-horse   DEPLOYED    alpine-0.1.0   1         default    Thu Dec  6 21:13:14 UTC
+```
+
+!!! tip:
+    The `gravity app` set of sub-commands support many of the same flags of
+    the respective `helm` commands such as `--set`, `--values`, `--namespace`
+    and so on. Check `--help` for each command to see which are supported.
+
+### Upgrade a Release
+
+To upgrade a release, build a new application image (say, `alpine-0.2.0.tar`),
+upload it to a cluster node and execute the upgrade command:
+
+```bsh
+$ gravity app upgrade falling-horse alpine-0.2.0.tar \
+    --registry 10.103.27.132:5000 \
+    --set image.registry=10.103.27.132:5000/
```

Comment:
is the `/` at the end mandatory? it looks weird in a non-HTTP address.

Reply:
maybe the CLI parser should be smart here?

Reply:
This goes directly to helm templating engine. Depending on how you define it in your template, e.g. in our case it's:

`{{ .Values.image.registry }}{{ .Values.image.repository }}:{{ .Values.image.tag }}`

so slash is required, otherwise registry and repo will "merge". On the other hand if I added slash to the template itself, then it wouldn't be possible to provide "empty" registry (cause there would be an extra slash at the front).